shared/ui/core/src/main/java/com/ivy/ui/FormatMoneyUseCase.kt

```diff
@@ -23,25 +32,69 @@ class FormatMoneyUseCase @Inject constructor(
     private val withoutDecimalFormatter = DecimalFormat("###,###", DecimalFormatSymbols(locale))
     private val withDecimalFormatter = DecimalFormat("###,###.00", DecimalFormatSymbols(locale))
     private val shortenAmountFormatter = DecimalFormat("###,###.##", DecimalFormatSymbols(locale))
+    private val cryptoFormatter =
+        DecimalFormat("###,###,##0.${"0".repeat(9)}", DecimalFormatSymbols(locale))
 
-    suspend fun format(value: Double, shortenAmount: Boolean): String {
-        if (abs(value) >= THOUSAND && shortenAmount) {
-            val result = if (abs(value) >= BILLION) {
+    /**
+     * Formats a currency or cryptocurrency amount based on the input parameters.
+     *
+     * @param value The numeric value to format.
+     * @param shortenAmount Flag to indicate if the amount should be shortened (e.g., "1k" for 1,000).
+     * @param isCrypto Flag to indicate if the value is a cryptocurrency, enabling up to 9 decimal places.
+     * @return The formatted string representation of the value.
+     */
+    suspend fun format(value: Double, shortenAmount: Boolean, isCrypto: Boolean = false): String {
+        val result = if (isCrypto) {
+            formatCrypto(value)
+        } else if (abs(value) >= THOUSAND && shortenAmount) {
+            if (abs(value) >= BILLION) {
                 "${shortenAmountFormatter.format(value / BILLION)}b"
             } else if (abs(value) >= MILLION) {
                 "${shortenAmountFormatter.format(value / MILLION)}m"
             } else {
                 "${shortenAmountFormatter.format(value / THOUSAND)}k"
             }
-            return result
         } else {
             val showDecimalPoint = features.showDecimalNumber.isEnabled(context)
 
             val formatter = when (showDecimalPoint) {
                 true -> withDecimalFormatter
                 false -> withoutDecimalFormatter
             }
-            return formatter.format(value)
+            formatter.format(value)
         }
+
+        return result
+    }
+
+    /**
+     * Formats a cryptocurrency value with up to 9 decimal places, removing unnecessary trailing zeros.
+     *
+     * @param value The cryptocurrency value to format.
+     * @return The formatted cryptocurrency value as a string.
+     */
+    private fun formatCrypto(value: Double): String {
+        val result = cryptoFormatter.format(value)
+        return when {
+            result.lastOrNull() == localDecimalSeparator().firstOrNull() -> {
```

Comment:
I don't like this logic - can we just fix the formatter?

Reply:
Removed the function, just relying on the suggested formatter now